game-core/src/main/java/games/strategy/triplea/ui/history/HistoryLog.java

```diff
@@ -402,8 +399,15 @@ private static boolean titleNeedsFurtherProcessing(final String title) {
   }
 
   @VisibleForTesting
-  static String parsePlayerNameFromDiceRollMessage(final String message) {
-    return message.contains(" roll ") ? message.substring(0, message.indexOf(" roll ")) : message;
+  static String parseHitDifferentialKeyFromDiceRollMessage(final String message) {
+    final Pattern diceRollPattern = Pattern.compile("^(\\w+) roll(?: (\\w+))? dice");
```

Comment:
Can player names contain spaces?  If so, this regex will need to be updated accordingly.

Reply:
I'm not aware of any limitation that they can't though I haven't seen any maps with a space in the player names.

Reply:
5e61e09a4 I went ahead and just allowed all characters in both the player name and dice type, as that matches (at least for the player name) the previous behavior.  Additional unit tests added, as required.